aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -347,13 +411,13 @@ sub pxe_append
     }
 }
 
-# Prints the PXE configuration file.
-sub pxeprint
+# Write the PXELINUX configuration file.
+sub write_pxelinux_config
 {
-    my $cfg = shift;
-    my $t = $cfg->getElement (PXEROOT)->getTree;
-    my $fh = CAF::FileWriter->open (filepath ($cfg),
-				    log => $this_app, mode => 0644);
+    my ($self, $cfg) = @_;
+    my $pxe_config = $cfg->getElement (PXEROOT)->getTree;
+    my $fh = CAF::FileWriter->open (filepath ($cfg, PXE_VARIANT_PXELINUX),
+                    log => $this_app, mode => 0644);
```

Comment:
Part of the WIP to turn every function into a method. Once done, I will use `$self` every where for the reporting.